File: fullstack/frontend/src/services/FeedbackService.js
import axios from "axios";

const API_URL = "http://localhost:8080/feedback";

const getAllFeedback = () => {
    return axios.get(`${API_URL}/all`);
};

const saveFeedback = (feedback) => {
    return axios.post(`${API_URL}/submit`, feedback);
};

const deleteFeedback = (id) => {
    return axios.delete(`${API_URL}/delete/${id}`);
};

export default {
    getAllFeedback,
    saveFeedback,
    deleteFeedback,
};

File: fullstack/fullstack/src/main/java/com/project/fullstack/service/FeedbackService.java
package com.project.fullstack.service;

import com.project.fullstack.entity.Feedback;
import com.project.fullstack.repository.FeedbackRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class FeedbackService {

    @Autowired
    private FeedbackRepository feedbackRepository;

    // Save Feedback
    public Feedback saveFeedback(Feedback feedback) {
        return feedbackRepository.save(feedback);
    }

    // Get All Feedback
    public List<Feedback> getAllFeedback() {
        return feedbackRepository.findAll();
    }

    // Delete Feedback
    public void deleteFeedback(Long id) {
        feedbackRepository.deleteById(id);
    }


    // update Feedback
    public Feedback updateFeedback(Long id, Feedback updatedFeedback) {

    Feedback feedback = feedbackRepository.findById(id)
            .orElseThrow(() -> new RuntimeException("Feedback not found"));

    feedback.setName(updatedFeedback.getName());
    feedback.setEmail(updatedFeedback.getEmail());
    feedback.setMessage(updatedFeedback.getMessage());

    return feedbackRepository.save(feedback);
}
}